### PR TITLE
add PSI and IBM cloud to the script for test

### PR DESCRIPTION
**Pull request by anandrkskd**
Signed-off-by: anandrkskd <anandrkskd@gmail.com>

**What type of PR is this?**

 /kind feature

**What does this PR do / why we need it**:

This PR adds script to run tests on Ibm cloud 

**Which issue(s) this PR fixes**:

Fixes - part of #4412
Fixes - part of #4408

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] Update changelog

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

**Comment by dharmit** [comment]
Where/how is the script `scripts/openshiftci-e2e-4x-psi-tests_win_mac.sh` going to be called? Is it going to be executed manually?

**Comment by anandrkskd** [comment]
> Where/how is the script `scripts/openshiftci-e2e-4x-psi-tests_win_mac.sh` going to be called? Is it going to be executed manually?

This particular file will be used in prow start the test flow with correct env var.
A similar example for better understanding is this [line](https://github.com/openshift/release/blob/b68cc78500eef7ea33eae529bb1ae3e62137f9b0/ci-operator/config/openshift/odo/openshift-odo-main__psi.yaml#L50) and [this](https://github.com/openshift/release/blob/b68cc78500eef7ea33eae529bb1ae3e62137f9b0/ci-operator/config/openshift/odo/openshift-odo-main__psi.yaml#L33)

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: scripts/run_script_e2e.sh [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    fi
+done
+
+# Generate random project names to some tests
+export REDHAT_OPENJDK11_RHEL8_PROJECT="${SCRIPT_IDENTITY}$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 4 | head -n 1)"
+export REDHAT_OPENJDK11_UBI8_PROJECT="${SCRIPT_IDENTITY}$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 4 | head -n 1)"
+export REDHAT_NODEJS12_RHEL7_PROJECT="${SCRIPT_IDENTITY}$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 4 | head -n 1)"
+export REDHAT_NODEJS12_UBI8_PROJECT="${SCRIPT_IDENTITY}$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 4 | head -n 1)"
+export REDHAT_NODEJS14_UBI8_PROJECT="${SCRIPT_IDENTITY}$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 4 | head -n 1)"
+
```
Comment (prietyc123) [inline scripts/run_script_e2e.sh:51]: Why are we not handling operator namespace here https://github.com/openshift/odo/blob/main/scripts/configure-installer-tests-cluster.sh#L23

File: scripts/run_script_e2e.sh [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+shout "Logging into 4x cluster as developer (logs hidden)"
+set +x
+if [[ $CLUSTER_TYPE == "PSI" ]]
+then
+    #PSI cluster login
+    oc login -u developer -p ${OCP4X_DEVELOPER_PASSWORD} --insecure-skip-tls-verify ${OCP4X_API_URL}
+else
+    # Login to IBM Cloud using service account API Key
+    ##ibmcloud login --apikey $IBMC_OCP47_APIKEY -a cloud.ibm.com -r eu-de -g "Developer CI and QE"
```
Comment (prietyc123) [inline scripts/run_script_e2e.sh:70]: same as https://github.com/openshift/odo/pull/4946#discussion_r689281584

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: scripts/run_script_e2e.sh (hunk 1 above)
Comment (anandrkskd) [reply scripts/run_script_e2e.sh:51]: As configuring cluster was one time thing, but as we are changing it to gingko taking care of installing we will remove this from the script.
As installing operator is for not only a one time setup we have not added it to the setup/run script

File: scripts/run_script_e2e.sh [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+shout "Logging into 4x cluster for some setup (logs hidden)"
+shout $CLUSTER_TYPE
+set +x
+if [[ $CLUSTER_TYPE == "PSI" ]]
+then
+    #PSI cluster login
+    oc login -u kubeadmin -p ${OCP4X_KUBEADMIN_PASSWORD} --insecure-skip-tls-verify  ${OCP4X_API_URL}
+else
+    # Login to IBM Cloud using service account API Key
+    ##ibmcloud login --apikey $IBMC_OCP47_APIKEY -a cloud.ibm.com -r eu-de -g "Developer CI and QE"
```
[the remark this answers is not in the document]
Comment (anandrkskd) [reply scripts/run_script_e2e.sh]: yes

File: scripts/run_script_e2e.sh [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    # Login to cluster in IBM Cloud using cluster API key
+    oc login --token=$IBMC_OCLOGIN_APIKEY --server=$IBMC_OCP47_SERVER
+  
+fi
+
+set -x
+
+# # Integration tests
+shout "Running integration Tests"
+#make test-operator-hub || error=true
```
Comment (prietyc123) [inline scripts/run_script_e2e.sh:81]: IIRC we are planning to run operatorhub as well on IBM or PSI, no?

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: scripts/run_script_e2e.sh (hunk 4 above)
Comment (anandrkskd) [reply scripts/run_script_e2e.sh:81]: yes, I will update it.

File: scripts/run_script_e2e.sh [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    # Login to cluster in IBM Cloud using cluster API key
+    oc login -u $IBM_OC_LOGIN_USER -p $IBMC_DEVELOPER_OCLOGIN_APIKEY --server=$IBMC_OCP47_SERVER
+  
+fi
+
+set -x
+
+# # Integration tests
+shout "Running integration Tests"
+make test-operator-hub || error=true
```
Comment (prietyc123) [inline scripts/run_script_e2e.sh:76]: @anandrkskd Could you please remove operator-hub test or comment it out. As you said we are facing issue https://github.com/openshift/odo/issues/5034 on macOS and win 

/lgtm cancel

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: scripts/run_script_e2e.sh (hunk 5 above)
Comment (anandrkskd) [reply scripts/run_script_e2e.sh:76]: @prietyc123 I have created a PR for that #5035 for that issue.
I think we can have operator-hub test in this PR

**Pull request merged**